### ISSUE TITLE
How to implement "early stop"

### ISSUE THREAD

**Issue by shaspear** [open-mmlab/mmdetection#9307]
### Discussed in https://github.com/open-mmlab/mmdetection/discussions/9304

<div type='discussions-op-text'>

<sup>Originally posted by **shaspear** November 11, 2022</sup>
Hi，
When I used mmdetection to train my data, there was an over fitting phenomenon, so I wanted to stop the training when the loss was minimum, but no relevant configuration item was found. Anyone can help me? thanks
</div>

**Comment by ZwwWayne** [comment]
It would be very easy if you choose mmdet 3.x. Basically, you can implement a train loop that check the validation loss every epoch and decide whether to stop

**Comment by sanbuphy** [comment]
These may be helpful to you:
CheckpointHook
https://mmengine.readthedocs.io/zh_CN/latest/tutorials/hook.html

https://mmdetection.readthedocs.io/en/3.x/user_guides/useful_hooks.html
[early stop]（https://microsoft.github.io/ai-edu/%E5%9F%BA%E7%A1%80%E6%95%99%E7%A8%8B/A2-%E7%A5%9E%E7%BB%8F%E7%BD%91%E7%BB%9C%E5%9F%BA%E6%9C%AC%E5%8E%9F%E7%90%86/%E7%AC%AC7%E6%AD%A5%20-%20%E6%B7%B1%E5%BA%A6%E7%A5%9E%E7%BB%8F%E7%BD%91%E7%BB%9C/16.4-%E6%97%A9%E5%81%9C%E6%B3%95.html）

**Comment by shaspear** [comment]
Thanks a lot

**Comment by tctco** [comment]
Hi @shaspear have you solved this problem? How should I break the training process in the middle?
